### PR TITLE
Add 'tzdata' package to Docker image

### PR DESCRIPTION
**Pull request by AltoNyan**
#### Description

Add `tzdata` package to Docker image by appending `tzdata` to Dockerfile's `apk add ...` section.

#### Checklist

 - [ ] I have updated tests where applicable.
 - [x] I have added an entry to the CHANGELOG.

### CODE REVIEW

**Comment by mhils** [comment]
Thanks for the PR! How does the lack of tzdata influence mitmproxy?

**Comment by AltoNyan** [comment]
Lack of `tzdata` makes we can't able to set a timezone in the Docker container.

Of course, mitmproxy working without a timezone well, but we can not sync timezone with host.

And, this makes some inconvenience with tasks like scheduling.

**Comment by mhils** [comment]
So this affects your particular project, but not mitmproxy users in general?

**Comment by AltoNyan** [comment]
No, I think timezone support is a common thing in Docker container, and this is why I make a pull request.

This is not for my particular project, I just think this is better.

Of course, if you think it's useless or unnecessary it's okay to ignore this PR.


ps. Make a pull request is the first time for me. So, I'm sorry if something is wrong or weird.

**Comment by mhils** [comment]
No worries, sorry if I've come off harsh! I'm just trying to understand the use case here. 😃 

Can you give me a concrete example of some use case that only works when tzdata is installed in the mitmproxy docker image? When/why do you want to set a timezone in a Docker container, and how? What kind of scheduling are you doing here?

**Comment by mhils** [comment]
Side note: we also want to switch from Alpine to a debian/ubuntu base image, probably https://hub.docker.com/_/python (#3712). Not sure what's already installed there.

**Comment by AltoNyan** [comment]
> No worries, sorry if I've come off harsh! I'm just trying to understand the use case here. 😃
> 
> Can you give me a concrete example of some use case that only works when tzdata is installed in the mitmproxy docker image? When/why do you want to set a timezone in a Docker container, and how? What kind of scheduling are you doing here?

Thanks for the kind reply. 😃 

Without tzdata, the Alpine Docker container cannot set timezone. So, its docker logs and the other time-related records are based on UTC only.
But, there is nothing precisely wrong or not working with it.  Just inconvenience and some efforts are needed to correct it.


When the first time I want to set timezone on the container is that I'm reading logs on the docker container.
Its timestamps are based on UTC. So, it's different from the current time.

Basically, there's two common way to sync timezone with the host.
1. Bind mount `/etc/localtime:/etc/localtime:ro`
2. Set `TZ` environment variable on docker run.

But it's not working with the Alpine-Linux image. (because lack of the `tzdata`). So I just install `tzdata` package inside of the container.

Schedule an update of some resources for cache based on days. And it has to convert its time based on UTC.

**Comment by AltoNyan** [comment]
> Side note: we also want to switch from Alpine to a debian/ubuntu base image, probably https://hub.docker.com/_/python (#3712). Not sure what's already installed there.


Wow, now I checked both images 'buster' and 'slim-buster' on https://hub.docker.com/_/python.

And, it perfectly works with the 'TZ' environmental variable to set timezone.

**Comment by mhils** [comment]
Thanks for the explanation! Let's focus on moving the image to `python:3.9-buster` then, PRs welcome!
